### PR TITLE
network,udev: move TransmitQueues=/ReceiveQueues= from .network to .link

### PR DESCRIPTION
**Pull request by yuwata**
See discussions in https://github.com/systemd/systemd/pull/18170#issuecomment-758807497 and https://github.com/orgs/systemd/teams/systemd/discussions/1.

### CODE REVIEW

**Comment by ssahani** [comment]
(that is, until someone shows up and can make a really good case that they want to switch the tx queue length depending on the network they connect to, but i#d wait until that happens, and then have the setting in both [Link] sections, in .link and in .network


Please add here not move.

**Comment by yuwata** [comment]
When a user really want them in .network file with a good reason, then let's also re-add them. I think no one requests now. Or, do you have any reason?

**Comment by ssahani** [comment]
> When a user really wants them in .network file with a good reason, then let's re-add them again. I think no one requests now. Or, do you have any reason?

No one requests because this is a new feature.  Ok

File: man/systemd.link.xml [hunk 1]
```diff
@@ -409,6 +409,20 @@
           be ignored.</para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>TransmitQueues=</varname></term>
+        <listitem>
+          <para>Specifies the devices's number of transmit queues. An integer in the range 1...4096.
```
Comment (Pesa) [inline man/systemd.link.xml]:
```suggestion
          <para>Specifies the device's number of transmit queues. An integer in the range 1..4096.
```

File: man/systemd.link.xml [hunk 2]
```diff
@@ -409,6 +409,20 @@
...
+          When unset, the kernel's default will be used.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>ReceiveQueues=</varname></term>
+        <listitem>
+          <para>Specifies the devices's number of receive queues. An integer in the range 1...4096.
```
Comment (Pesa) [inline man/systemd.link.xml]:
```suggestion
          <para>Specifies the device's number of receive queues. An integer in the range 1..4096.
```

**Comment by yuwata** [comment]
@Pesa Thank you!

File: man/systemd.link.xml [hunk 3]
```diff
@@ -410,7 +410,7 @@
         </listitem>
       </varlistentry>
       <varlistentry>
-        <term><varname>TxQueueLength=</varname></term>
+        <term><varname>TransmitQueueLength=</varname></term>
```
Comment (poettering) [inline man/systemd.link.xml:427]: "Tx" is one of those abbreviations i'd actually be OK with. It's very very standard I'd say. But Transmit is good too, and internal consistency matters most I guess

**Review by poettering — APPROVED** [review]

**Comment by poettering** [comment]
lgtm, but needs rebase

File: man/systemd.link.xml (hunk 3 above)
Comment (yuwata) [reply man/systemd.link.xml:427]: Yeah, but we already have e.g. `TransmitChecksumOffload=`. Just for consistency.

**Comment by yuwata** [comment]
@poettering Thank you for your review. Rebased. Setting the green label.

**Pull request merged**